server/protocols/jmap/src/test/java/org/apache/james/jmap/model/MessageFactoryTest.java

```diff
@@ -213,7 +219,13 @@ public void headersShouldBeUnfoldedAndDecoded() throws Exception {
             .preview("(Empty)")
             .textBody(Optional.of(""))
             .htmlBody(Optional.empty())
+            .isAnswered(false)
```

Comment:
Maybe a ".flags(Flags flags)" method would be more convenient...

Here values of "is\*" properties and .keywords is information duplication.... By the way, what happens in case of mismatch?